native/cocos/renderer/pipeline/custom/NativePipelineTypes.h

```diff
@@ -1560,6 +1597,29 @@ class NativePipeline final : public Pipeline {
     void resetRenderQueue(bool reset) override;
     bool isRenderQueueReset() const override;
 
+    ccstd::string getName() const override;
+    void setName(const ccstd::string &name) override;
+    void setCustomBehavior(const ccstd::string &name) override;
+
+    void setMat4(const ccstd::string &name, const Mat4 &mat) override;
+    void setQuaternion(const ccstd::string &name, const Quaternion &quat) override;
+    void setColor(const ccstd::string &name, const gfx::Color &color) override;
+    void setVec4(const ccstd::string &name, const Vec4 &vec) override;
+    void setVec2(const ccstd::string &name, const Vec2 &vec) override;
```

Comment:
Doesn't need `setVec3`?

Reply:
> Doesn't need `setVec3`?

`Vec3` is very error prone for buffer layout alignment calculation. We suggest using `Vec4` instead.